src/cart_cell.h

```diff
@@ -28,127 +31,382 @@
 namespace bdm {
 
 
-// ─────────────────────────────
-// CartCellState Enum Definition
-// ─────────────────────────────
+/**
+ * @brief Enumeration defining the possible states of a CAR-T cell
+ * 
+ * This enum class represents the different states that a CAR-T cell can be in
+ * during its lifecycle in the simulation.
+ */
 enum class CartCellState : int {
-  //living cell
+  /** @brief Living cell state - the cell is alive and functioning normally */
   kAlive=0,
 
-  // Apoptotic schedule of a apoptotic cell: controled death
-  kApoptotic=1 // Apoptotic phase (the cell is undergoing programmed cell death characterized by cell shrinkage)
+  /** @brief Apoptotic phase - the cell is undergoing programmed cell death
+   *  characterized by cell shrinkage and controlled death
+   */
+  kApoptotic=1
 };
 
-// ─────────────────────────────
-// CartCell Class Definition
-// ─────────────────────────────
+/**
+ * @brief CAR-T cell class implementation
+ * 
+ * This class represents a CAR-T (Chimeric Antigen Receptor T-cell) in the simulation.
+ * It inherits from the base Cell class and includes specific behaviors and properties
+ * related to CAR-T cell biology, including states, volume dynamics, and interactions
+ * with tumor cells.
+ */
 class CartCell : public Cell {
   BDM_AGENT_HEADER(CartCell, Cell, 1);
 
   public:
+  /** @brief Default constructor */
```

Comment:
We don't need that level of documentation. Only the things that are less obvious.